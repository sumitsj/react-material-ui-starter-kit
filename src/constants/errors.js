export const ERROR_TYPES = {
  SYSTEM_ERROR: 'systemError',
  PAGE_NOT_FOUND_ERROR: 'pageNotFoundError',
};

export const ERROR_MESSAGES = {
  systemError: {
    title: 'SYSTEM ERROR',
    message: '',
  },
  pageNotFoundError: {
    title: 'PAGE NOT FOUND',
    message: 'May be the page you are looking for has been removed, or you typed in the wrong URL',
  },
};
