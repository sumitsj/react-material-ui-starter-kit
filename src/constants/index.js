export { ERROR_TYPES, ERROR_MESSAGES } from './errors';
