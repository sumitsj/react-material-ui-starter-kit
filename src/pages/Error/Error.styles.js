import makeStyles from '@material-ui/core/styles/makeStyles';

const useStyles = makeStyles((theme) => ({
  container: {
    marginTop: '20vh',
  },
  logo: {
    fontSize: '6rem',
  },
}));

export default useStyles;
