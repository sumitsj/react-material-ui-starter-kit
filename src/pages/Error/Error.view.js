import React from 'react';
import * as PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import ErrorOutlineOutlinedIcon from '@material-ui/icons/ErrorOutlineOutlined';
import { ERROR_MESSAGES, ERROR_TYPES } from '../../constants';
import Header from '../../components/Header';
import useStyles from './Error.style';

const Error = ({ type }) => {
  const classes = useStyles();

  return (
    <Container maxWidth="lg" className={classes.container} align="center">
      <Header />
      <ErrorOutlineOutlinedIcon className={classes.logo} />
      <Typography variant="h4" gutterBottom>
        {ERROR_MESSAGES[type].title}
      </Typography>

      <Typography variant="subtitle1" gutterBottom>
        {ERROR_MESSAGES[type].message}
      </Typography>
    </Container>
  );
};

Error.propTypes = {
  type: PropTypes.string,
};

Error.defaultProps = {
  type: ERROR_TYPES.PAGE_NOT_FOUND_ERROR,
};

export default Error;
