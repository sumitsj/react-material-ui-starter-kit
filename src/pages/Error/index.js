import Error from './Error.view';

export default Error;
