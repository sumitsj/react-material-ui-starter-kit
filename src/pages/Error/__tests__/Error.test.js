import React from 'react';
import { render } from '@testing-library/react';
import Error from '..';
import { ERROR_TYPES } from '../../../constants';

jest.mock('../../../components/Header', () => jest.fn(() => <header />));

describe('Error', () => {
  describe('HTML Structure', () => {
    it('should render properly with default props', () => {
      const { asFragment } = render(<Error />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render properly with custom props', () => {
      const { asFragment } = render(<Error type={ERROR_TYPES.SYSTEM_ERROR} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
