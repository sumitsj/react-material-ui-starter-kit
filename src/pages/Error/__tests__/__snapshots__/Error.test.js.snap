// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error HTML Structure should render properly with custom props 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiContainer-root makeStyles-container-50 MuiContainer-maxWidthLg"
  >
    <header />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-logo-51"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
    >
      SYSTEM ERROR
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
    />
  </div>
</DocumentFragment>
`;

exports[`Error HTML Structure should render properly with default props 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthLg"
  >
    <header />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-logo-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
    >
      PAGE NOT FOUND
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
    >
      May be the page you are looking for has been removed, or you typed in the wrong URL
    </h6>
  </div>
</DocumentFragment>
`;
